### PR TITLE
Improved evaluation memory and speed efficiency in NER

### PR DESCRIPTION
**Pull request by mhmda**
Made evaluation more efficient time and memory wise.

Previously, all the predictions were appended after each batch and in the end the argmax is taken. Instead, we can just take an argmax for each batch and keep appending it saving both time and memory. Even in simple cases where we only have two tags, the new `preds` will be 50% smaller than the original `preds`. In my case (10,000+ tags) the original `evaluate` does not work and always throws a `MemoryError` even on a server with over 100GB of memory.

### CODE REVIEW

File: simpletransformers/ner/ner_model.py [hunk 1]
```diff
@@ -726,7 +726,6 @@ def evaluate(self, eval_dataset, output_dir, verbose=True, silent=False, wandb_l
 
         eval_loss = eval_loss / nb_eval_steps
         token_logits = preds
```
Comment (ThilinaRajapakse) [inline simpletransformers/ner/ner_model.py:728]: Wouldn't making this change mean that we'll lose token logits here?

Comment (mhmda) [reply simpletransformers/ner/ner_model.py:728]: Sorry, missed that. That's certainly an issue. I think that having a fast evaluate function that only does evaluation would be super useful. What about adding a new argument to `evaluate` (e.g. `return_logits=False`) to determine if the logits should be returned, and if not `results, None , preds_list` will be returned instead?

Comment (ThilinaRajapakse) [reply simpletransformers/ner/ner_model.py:728]: I can see the value in that. Perhaps both the `evaluate` and the `predict` functions should have the `return_logits` flag?

Another issue here is that `np.append()` is being called in the loop. Creating the array beforehand and inserting the values in the loop is more efficient. I've already done this for the `ClassificationModel`, and plan on doing it for `NER` as well.

Comment (mhmda) [reply simpletransformers/ner/ner_model.py]: - Added `return_logits` flag to `evaluate` and `eval_model`.
- Created the predictions array before the loop instead of relying on `np.append()`.

**Comment by mhmda** [comment]
- Added `return_logits` flag to `evaluate` and `eval_model` in `ner_model`.
- Created the predictions array before the loop instead of relying on `np.append()`.
